implementation/jwt-build/src/main/java/io/smallrye/jwt/build/impl/JwtClaimsBuilderImpl.java

```diff
@@ -265,7 +265,7 @@ public JwtEncryptionBuilder innerSign() throws JwtSignatureException {
      */
     @Override
     public JwtEncryptionBuilder jwe() {
-        JwtBuildUtils.setDefaultJwtClaims(claims, tokenLifespan);
+        lastJti = JwtBuildUtils.setDefaultJwtClaims(claims, lastJti, tokenLifespan);
         return new JwtEncryptionImpl(claims.toJson());
     }
```

Comment:
Would it make sense to instead remove the `jti` claim from the `JwtClaims` before returning rather than keeping track of the previous value? Just mentioning in case it hasn't been considered.

Reply:
Hi Mike @MikeEdgar It might've crossed my mind but I did not look into it. May be it would be a cleaner solution. It might not be possible to ensure that in all the cases the uniqueness is assured, so perhaps just removing it would do; I'll have a look.

Reply:
@MikeEdgar I've fixed it to apply your idea, which is simpler and easier to understand, the original approach I took was looking not good at all to me this morning when I returned to changing it :-), thanks